### ISSUE TITLE
defercheck false positives

### ISSUE THREAD

**Issue by robfig** [opennota/check#6]
Great idea for the tools!

I ran defercheck on my codebase and found 2 false positives, which I thought you may be interested to know, in case the check could be tightened.  Feel free to close this issue if it's not possible to recognize these cases

Case 1 - Inline goroutines call WaitGroup.Done()
```go

if doStep {
  wg.Add(1)
  go func() { defer wg.Done(); ... }()
}
if doOtherStep {
  wg.Add(1)
  go func() { defer wg.Done(); ... }()
}
```

Case 2 - Reassignment so that the defer calls are not on the same receiver
```go
        fileReader, _ := bucket.GetReader(file)
        defer fileReader.Close()
        if strings.HasSuffix(file, ".gz") {
                fileReader, _ = gzip.NewReader(fileReader)
                defer fileReader.Close()
        }

       .. read file contents from fileReader ...
```

Thanks again

**Comment by opennota** [comment]
Can you please check whether the problem still exists?
